CS8981 (The type name only contains lower-cased ascii characters.) when generating C# code with names=original

When compiling for C# 11, this warning may appear:
>  warning CS8981: The type name 'keyvalue' only contains lower-cased ascii characters. Such names may become reserved for the language. 

here: `public partial class keyvalue : global::ProtoBuf.IExtensible`

It's not necessarily wrong, but protobuf-gen already has an `Escape` method for all known language identifiers, so this doesn't seem like that big of a problem, perhaps this warning should be suppressed?

Mildly related: https://github.com/protocolbuffers/protobuf/pull/9981

--- a/src/Examples/Issues/Issue647.Generated.cs
+++ b/src/Examples/Issues/Issue647.Generated.cs
@@ -5,7 +5,7 @@
 // </auto-generated>
 
 #region Designer generated code
-#pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 namespace ProtoBuf.Issues.Issue647Types.Bar
 {
 
@@ -54,5 +54,5 @@ namespace ProtoBuf.Issues.Issue647Types.Foo
 
 }
 
-#pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 #endregion

--- a/src/Examples/Issues/Issue855Proto2CSharp.Generated.cs
+++ b/src/Examples/Issues/Issue855Proto2CSharp.Generated.cs
@@ -5,7 +5,7 @@
 // </auto-generated>
 
 #region Designer generated code
-#pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 [global::ProtoBuf.ProtoContract()]
 public partial class BarClass : global::ProtoBuf.IExtensible
 {
@@ -128,5 +128,5 @@ public partial class BarClass : global::ProtoBuf.IExtensible
 
 }
 
-#pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 #endregion

--- a/src/Examples/Issues/Issue855Proto3CSharp.Generated.cs
+++ b/src/Examples/Issues/Issue855Proto3CSharp.Generated.cs
@@ -5,7 +5,7 @@
 // </auto-generated>
 
 #region Designer generated code
-#pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 [global::ProtoBuf.ProtoContract()]
 public partial class BarClass : global::ProtoBuf.IExtensible
 {
@@ -58,5 +58,5 @@ public partial class BarClass : global::ProtoBuf.IExtensible
 
 }
 
-#pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 #endregion

--- a/src/protobuf-net.Reflection/CSharpCodeGenerator.cs
+++ b/src/protobuf-net.Reflection/CSharpCodeGenerator.cs
@@ -141,7 +141,7 @@ namespace ProtoBuf.Reflection
                .WriteLine("// </auto-generated>")
                .WriteLine()
                .WriteLine("#region Designer generated code")
-               .Write($"#pragma warning disable {prefix}0612, {prefix}0618, {prefix}1591, {prefix}3021");
+               .Write($"#pragma warning disable {prefix}0612, {prefix}0618, {prefix}1591, {prefix}3021, {prefix}8981");
             if (ctx.Supports(CSharp6))
             {
                 tw.Write(AdditionalSuppressionCodes);
@@ -168,7 +168,7 @@ namespace ProtoBuf.Reflection
         protected override void WriteFileFooter(GeneratorContext ctx, FileDescriptorProto file, ref object state)
         {
             var prefix = ctx.Supports(CSharp6) ? "CS" : "";
-            var tw = ctx.Write($"#pragma warning restore {prefix}0612, {prefix}0618, {prefix}1591, {prefix}3021");
+            var tw = ctx.Write($"#pragma warning restore {prefix}0612, {prefix}0618, {prefix}1591, {prefix}3021, {prefix}8981");
             if (ctx.Supports(CSharp6))
             {
                 tw.Write(AdditionalSuppressionCodes);

--- a/src/protobuf-net.Reflection/Descriptor.cs
+++ b/src/protobuf-net.Reflection/Descriptor.cs
@@ -5,7 +5,7 @@
 // </auto-generated>
 
 #region Designer generated code
-#pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 namespace Google.Protobuf.Reflection
 {
 
@@ -1322,5 +1322,5 @@ namespace Google.Protobuf.Reflection
 
 }
 
-#pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 #endregion

--- a/src/protobuf-net.Test/NanoPBOptions.cs
+++ b/src/protobuf-net.Test/NanoPBOptions.cs
@@ -8,7 +8,7 @@
 // </auto-generated>
 
 #region Designer generated code
-#pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 namespace NanoPB
 {
     [global::ProtoBuf.ProtoContract()]
@@ -396,5 +396,5 @@ namespace NanoPB
 
     }
 }
-#pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+#pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
 #endregion

--- a/src/protobuf-net.Test/Nullables/WrappersProto/WrappersProto.ProtoToCode.cs
+++ b/src/protobuf-net.Test/Nullables/WrappersProto/WrappersProto.ProtoToCode.cs
@@ -76,7 +76,7 @@ namespace ProtoBuf.Test.Nullables.WrappersProto
             // </auto-generated>
 
             #region Designer generated code
-            #pragma warning disable CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+            #pragma warning disable CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
             [global::ProtoBuf.ProtoContract()]
             public partial class WrappedTest : global::ProtoBuf.IExtensible
             {{
@@ -90,7 +90,7 @@ namespace ProtoBuf.Test.Nullables.WrappersProto
 
             }}
 
-            #pragma warning restore CS0612, CS0618, CS1591, CS3021, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
+            #pragma warning restore CS0612, CS0618, CS1591, CS3021, CS8981, IDE0079, IDE1006, RCS1036, RCS1057, RCS1085, RCS1192
             #endregion
         ";
     }

